### PR TITLE
Add unittest tizen 7.0

### PR DESCRIPTION
**Pull request by lhs8928**
Commit 1:     [unittest] add negative unittest cases
    
    - added negative unittest cases
    - deleted unittest cases that cause GTEST.MEANINGLESS_ASSERTION defect
    
    **Self evaluation:**
    1. Build test:   [X]Passed [ ]Failed [ ]Skipped
    2. Run test:     [X]Passed [ ]Failed [ ]Skipped
    
    Signed-off-by: Seungbaek Hong <sb92.hong@samsung.com>

Commit 2:     [Test] Add unit tests for newly added tensor type
    
    This PR adds unit tests for the recently implemented tensor data types.
    The purpose is to ensure their functionality and correctness by performing various test cases.
    
    **Changes proposed in this PR:**
    - Added unit tests to test functions for CharTensor and ShortTensor
    - Check if both tensor data are contiguous before copy()
    
    **Self-evaluation:**
    1. Build test: [X]Passed [ ]Failed [ ]Skipped
    2. Run test:   [X]Passed [ ]Failed [ ]Skipped
    
    Signed-off-by: Donghyeon Jeong <dhyeon.jeong@samsung.com>

Commit 3:     [Unittest] Add network_graph unit test
    
    Add More unit test on network_graph
    - add reinitialize
    - catch exceptions
    
    **Self evaluation:**
    1. Build test:   [X]Passed [ ]Failed [ ]Skipped
    2. Run test:     [X]Passed [ ]Failed [ ]Skipped
    
    Signed-off-by: Donghak PARK <donghak.park@samsung.com>

Commit 4:     [Loss] Remove Empty KLD Loss files
    
    Currently, the KLD Loss file has only the function name and is empty.
    Therefore, I will remove this file now and implement it after retesting with test cases and verification in the next issue #2757.
    
    **Self evaluation:**
    1. Build test:   [X]Passed [ ]Failed [ ]Skipped
    2. Run test:     [X]Passed [ ]Failed [ ]Skipped
    
    Signed-off-by: Donghak PARK <donghak.park@samsung.com>

Commit 5:     [Unittest] re-activate the deactivated unittest
    
    I have reactivated the deactivated unittest except for the test
    cases that cause errors.
    
    - enable mol_attention_layer unittest
    - enable reduce_mean_layer unittest
    - enable unittest_models except for test cases that cause errors
    
    **Self evaluation:**
    1. Build test:   [X]Passed [ ]Failed [ ]Skipped
    2. Run test:     [X]Passed [ ]Failed [ ]Skipped
    
    Signed-off-by: Seungbaek Hong <sb92.hong@samsung.com>

Commit 6:     [Unittest] Add constant DerivativeLossLayer unittest
    
    Currently, there is a layersemantics in the NNTrainer Unittest that verifies the behavior of the losslayer
    but it does not include the case for the constant DerivativeLossLayer.
    
    By simply filling in the parameters and passing them over, we can perform unit tests on basic operations
    so I added this feature.
    
    **Self evaluation:**
    1. Build test:   [X]Passed [ ]Failed [ ]Skipped
    2. Run test:     [X]Passed [ ]Failed [ ]Skipped
    
    Signed-off-by: Donghak PARK <donghak.park@samsung.com>

Commit 7:     [Tensor] Parameterized constructor with TensorBase
    
    This PR updates a parameterized constructor accepting a shared pointer of the TensorBase class.
    Users now have greater flexibility when creating Tensors, enabling them to adapt new types of user-defined tensors by initializing the itensor object.
    Please note that TensorBase remains an abstract class and cannot be instantiated directly.
    Remember to utilize a shared_ptr with a derived class when employing this constructor.
    
    **Self-evaluation:**
    1. Build test: [X]Passed [ ]Failed [ ]Skipped
    2. Run test:   [X]Passed [ ]Failed [ ]Skipped
    
    Signed-off-by: Donghyeon Jeong <dhyeon.jeong@samsung.com>

Commit 8:     [UnitTest] Add kld-loss unit test
    
    Currently, there is no unit test for the KLD loss. Therefore, I am adding a new unit test to increase code coverage.
    - use pre defined unit test
    - (LayerSemanticsParamType)
    
    **Self evaluation:**
    1. Build test:   [X]Passed [ ]Failed [ ]Skipped
    2. Run test:     [X]Passed [ ]Failed [ ]Skipped
    
    Signed-off-by: Donghak PARK <donghak.park@samsung.com>

Commit 9:     [UnitTest] Add integration_tests on unit test
    
    In the case of NNTrainer, there are many instances where one function works in response to another function, so we have created integration tests for unit testing.
    - There are exceptional cases depending on the activation when it comes to losses.
    - For instance with BN (Batch Normalization), you can compare accuracy after conducting several epochs of testing.
    
    **Self evaluation:**
    1. Build test:   [X]Passed [ ]Failed [ ]Skipped
    2. Run test:     [X]Passed [ ]Failed [ ]Skipped
    
    Signed-off-by: Donghak PARK <donghak.park@samsung.com>

Commit 10:     [Loss] Implement KLD Loss Layer
    
    Implement KLD Loss on NNtrainer
    
    **Self evaluation:**
    1. Build test:   [X]Passed [ ]Failed [ ]Skipped
    2. Run test:     [X]Passed [ ]Failed [ ]Skipped
    
    Signed-off-by: Donghak PARK <donghak.park@samsung.com>

Commit 11:     [Trivial] apply clang-format for unittest
    
    Apply coding style with clang-format ( unittest files )
    
    **Self evaluation:**
    1. Build test:   [X]Passed [ ]Failed [ ]Skipped
    2. Run test:     [X]Passed [ ]Failed [ ]Skipped
    
    Signed-off-by: Donghak PARK <donghak.park@samsung.com>

Commit 12:     [Loss] Add finalize to KLD Loss
    
    Implement Finalize on KLD Loss
    
    **Self evaluation:**
    1. Build test:   [X]Passed [ ]Failed [ ]Skipped
    2. Run test:     [X]Passed [ ]Failed [ ]Skipped
    
    Signed-off-by: Donghak PARK <donghak.park@samsung.com>

Commit 13:     [Unnittest & Loss] Update KLD Loss & Fix Unittest
    
    Update KLD Loss Function Reflect review
    
    **Self evaluation:**
    1. Build test:   [X]Passed [ ]Failed [ ]Skipped
    2. Run test:     [X]Passed [ ]Failed [ ]Skipped
    
    Signed-off-by: Donghak PARK <donghak.park@samsung.com>

Commit 14:     [model] remove unused functions
    
     - Remove addLayerNode, getUnsortedLayers, setLoss, rotate_180
    
    Signed-off-by: hyeonseok <hs89.lee@samsung.com>

Commit 15:     [test] add model unittest
    
     - Added model api unittest
    
    Signed-off-by: hyeonseok <hs89.lee@samsung.com>

Commit 16:     [layer] remove mol attention
    
     - Temporarily remove mol attention layer
    
    Signed-off-by: hyeonseok <hs89.lee@samsung.com>

Commit 17:     [layer] remove time dist
    
     - Temporarily remove time dist layer
    
    Signed-off-by: hyeonseok <hs89.lee@samsung.com>

Commit 18:     [test] add optimized v2 & v3 planner unittest
    
     - Added optimized v2 & v3 planner parameterized unittest
    
    Signed-off-by: hyeonseok <hs89.lee@samsung.com>

Commit 19:     [test] added util test case
    
     - Added absFloat, exp_util unit test case
    
    Signed-off-by: hyeonseok <hs89.lee@samsung.com>

### CODE REVIEW

**Review by jijoongmoon — APPROVED** [review]
LGTM

**Review by baek2sm — APPROVED** [review]
LGTM

**Review by DonghakPark — APPROVED** [review]
LGTM!
